src/analytics/dsl.tsx

```diff
@@ -0,0 +1,12 @@
+export const dsl = {
+  "Follow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
+  },
+  "Unfollow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
+  },
+}
```

Comment:
We should also think about how we can best add typings to those props. Do we want to reference specific components for these analytics event hooks? If so, we could maybe do something like:

```js
import { for, on } from "./analytics"

import Artist from "../containers/artist"

for(Artist, artist => {
  artist.on("Follow Artist", {
    intent: (object) => ({ artist_id: object.artist.id }),
    success: (object) => ({ artist_id: object.artist.id }),
    fail: (object, error) => ({ artist_id: object.artist.id }),
  })
})
```

That way we can use generics to get the props interface from the component and type the `object` yielded to each function as the same props type.